profanity.rb

```diff
@@ -799,7 +799,7 @@ def get_color_pair_id(fg_code, bg_code)
 	str = str.gsub('lines', Curses.lines.to_s).gsub('cols', Curses.cols.to_s)
 	str.untaint
 	begin
-		proc { $SAFE = 3; eval(str) }.call.to_i
+		proc { $SAFE = 1; eval(str) }.call.to_i
```

Comment:
The SAFE setting does not affect the entire thread.  It only affects the proc, which is the only reason a proc is created and instantly called.

The limits I was interested in with safe level 3 is no arbitrary disk access, socket access, and executing programs.  Safe level 1 doesn't do these things, and as far as I know it doesn't do anything else for this situation since the string being evaled is already untainted.  Both the safe level and the proc can be removed.

While I'm sure SAFE has its issues, by removing it we've gone from needing an experienced Ruby programmer to turn a config file into a virus, to needing a toddler to do it.  Given the user unfriendliness of the config file, I don't think it's unreasonable to expect some users to share their config files, putting them at risk.

But, Ruby < 2.3 is only going to become more rare as time goes on, so SAFE has to go.  As an alternative, perhaps just gsubing out everything except "lines", " rows", numbers and math symbols before evaling would he sufficient.